pallets/xcmp-handler/src/lib.rs

```diff
@@ -434,15 +524,30 @@ pub mod pallet {
 		/// Pay for XCMP fees.
 		/// Transfers fee from payer account to the local chain sovereign account.
 		///
-		pub fn pay_xcm_fee(source: T::AccountId, fee: u128) -> Result<(), DispatchError> {
+		pub fn pay_xcm_fee(
+			currency_id: T::CurrencyId,
+			source: T::AccountId,
+			fee: u128,
+		) -> Result<(), DispatchError> {
 			let local_sovereign_account =
 				Sibling::from(T::SelfParaId::get()).into_account_truncating();
 
-			match T::Currency::transfer(
+			ensure!(
```

Comment:
Do we really need this? At the moment that code is only called after we have already checked that this action won't kill the account. I know by itself it should use this logic, but we could leave this out and update the ORML repo to have a transfer that does not allow death to eventually be used here

Reply:
I was doing it out of abundant caution but it is redundant with the current code structure.